src/sources/prometheus/scrape.rs

```diff
@@ -90,6 +93,37 @@ impl SourceConfig for PrometheusScrapeConfig {
             .endpoints
             .iter()
             .map(|s| s.parse::<http::Uri>().context(sources::UriParseSnafu))
+            .map(|r| {
+                if let Ok(uri) = r {
+                    let mut serializer = url::form_urlencoded::Serializer::new(String::new());
+                    if let Some(query) = uri.query() {
+                        for (k, v) in url::form_urlencoded::parse(query.as_bytes()) {
+                            serializer.append_pair(&k, &v);
+                        }
+                    };
```

Comment:
```suggestion
                    let mut serializer = url::form_urlencoded::Serializer::new(String::new());
                    if let Some(query) = uri.query() {
                        serializer.extend_pairs(url::form_urlencoded::parse(query.as_bytes()))
                    }
```

Nit, I think this could be written like this?

Reply:
Ha! I knew there was a better way to do this.
I read about `extended_pairs()` in my initial research but then I forgot of its existence hehe.
Thanks for the suggestion!